tailscale.tf

```diff
@@ -0,0 +1,8 @@
+# Generate Tailscale auth key
+resource "tailscale_tailnet_key" "this" {
+  reusable      = true
+  ephemeral     = true
+  preauthorized = true
+  expiry        = var.tailscale_key_expiry
+  tags          = var.tailscale_tags
```

Comment:
I would use this opportunity to add default tags, such as `tag:${var.env}` and then add concat var.tags to it.

If you want you can use locals for that purpose too.